packages/teleport-plugin-angular-base-component/src/utils.ts

```diff
@@ -2,24 +2,53 @@ import * as types from '@babel/types'
 import {
   convertValueToLiteral,
   getTSAnnotationForType,
+  createMethodsObject,
 } from '@teleporthq/teleport-shared/dist/cjs/utils/ast-js-utils'
-import { UIDLPropDefinition, UIDLStateDefinition } from '@teleporthq/teleport-types'
+import {
+  UIDLPropDefinition,
+  UIDLStateDefinition,
+  UIDLEventHandlerStatement,
+} from '@teleporthq/teleport-types'
 
 export const generateExportAST = (
   componentName: string,
   propDefinitions: Record<string, UIDLPropDefinition>,
   stateDefinitions: Record<string, UIDLStateDefinition>,
   dataObject: Record<string, any>,
+  methodsObject: Record<string, UIDLEventHandlerStatement[]>,
   t = types
 ) => {
-  const propDeclaration = Object.keys(propDefinitions).map((propKey) =>
-    t.classProperty(
+  let angularMethodsAST = []
+  if (Object.keys(methodsObject).length > 0) {
+    angularMethodsAST = createMethodsObject(
+      methodsObject,
+      propDefinitions,
+      'angular'
+    ) as types.ClassMethod[]
+  }
+
+  const propDeclaration = Object.keys(propDefinitions).map((propKey) => {
+    const definition = propDefinitions[propKey]
+    if (definition.type === 'func') {
+      return t.classProperty(
```

Comment:
@alexnm  what is your opinion. I am not sure if it is a safe assumption to consider all props which are functions are event-based ones.

Repl Link --> https://repl.teleporthq.io/?uidlLink=qoaic5fuhhl6w8hgej185h
Code sandbox link --> https://codesandbox.io/s/angular-6zw59

Reply:
I think this works for now, please write a comment about this as well, so we know in the future